docs/proposals/guidelines/proposal-template.md

```diff
@@ -0,0 +1,212 @@
+---
+title: neat-enhancement-idea
+authors:
+  - "@janedoe"
+reviewers:
+  - TBD
+  - "@alicedoe"
+approvers:
+  - TBD
+  - "@oscardoe"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/docs/proposals/this-other-neat-thing.md"  
+replaces:
+  - "/docs/proposals/that-less-than-great-idea.md"
+superseded-by:
+  - "/docs/proposals/our-past-effort.md"
+---
+
+# Neat Enhancement Idea
+
+This is the title of the enhancement. Keep it simple and descriptive. A good title can help
+communicate what the enhancement is and should be considered as part of any review.
+
+The YAML `title` should be lowercased and spaces/punctuation should be replaced with `-`.
+
+To get started with this template:
+
+1. **Make a copy of this template.** Copy this template into the main `proposals` directory.
+2. **Fill out the "overview" sections.** This includes the Summary and Motivation sections. These
+   should be easy and explain why the community should desire this enhancement.
+3. **Create a PR.** Assign it to folks with expertise in that domain to help sponsor the process.
```

Comment:
In our enhancement proposal, we need to specify very clearly, that a PR to implement the proposal should be drafted only after the proposal is accepted.
Otherwise, what is the need of the proposal? 

There would be these huge PRs with feature changes... and it will take a lot of time to get them in.. which can in turn spoil the contribution experience.

So, draft a PR only after the proposal is accepted.

Do you have any concerns with this @sbose78 ?

Reply:
As previously discussed, folks are free to open a PR anytime they want and show us how the implementation looks, but could still be asked for an enhancement proposal if there is a need for one.